src/languages/sql.js

```diff
@@ -1,168 +1,647 @@
 /*
  Language: SQL
- Contributors: Nikolay Lisienko <info@neor.ru>, Heiko August <post@auge8472.de>, Travis Odom <travis.a.odom@gmail.com>, Vadimtro <vadimtro@yahoo.com>, Benjamin Auder <benjamin.auder@gmail.com>
  Website: https://en.wikipedia.org/wiki/SQL
- Category: common
+ Category: common, database
  */
 
+/*
+
+Goals:
+
+SQL is intended to highlight basic/common SQL keywords and expressions
+
+- If pretty much every single SQL server includes supports, then it's a canidate.
+- It is NOT intended to include tons of vendor specific keywords (Oracle, MySQL,
+  PostgreSQL) although the list of data types is purposely a bit more expansive.
+- For more specific SQL grammars please see:
+  - PostgreSQL and PL/pgSQL - core
+  - T-SQL - https://github.com/highlightjs/highlightjs-tsql
+  - sql_more (core)
+
+ */
+
+import * as regex from '../lib/regex.js';
+
 export default function(hljs) {
-  var COMMENT_MODE = hljs.COMMENT('--', '$');
+  const COMMENT_MODE = hljs.COMMENT('--', '$');
+  const STRING = {
+    className: 'string',
+    variants: [
+    {
+      begin: '\'', end: '\'',
+      contains: [{begin: '\'\''}]
+    },
+    {
+      begin: '"', end: '"',
+      contains: [{begin: '""'}]
+    },
```

Comment:
Double quoted text is not a string, it is an identifier. Like `CREATE TABLE "my table" ( ...)`.

Reply:
Wait, what...

```
select "hey what is this if not a stirng"
```

Is that not a string?  We don't have a CSS class for 'identifer'... we have `variable`.  But you're just confusing me here.  Or are you saying in SOME cases it's an identifier... because right now at first glance it seems "ok" to just highlight all these as strings.

Reply:
It is _not_ a string (:

```
postgres=# select "hey what is this if not a stirng";
ERROR:  column "hey what is this if not a stirng" does not exist
```

The same is true for Oracle (and from the standard's perspective).
AFAIK MySQL accepts strings in double quotes. and MSSQL has a switch for this behavior.

Reply:
It is an identifier, so it shouldn't be highlighted.

Reply:
Ah, you're saying it's another MySQL ism...

```sql
-- identifier
select "hey what is this if not a stirng";
-- string
select 'hey what is this if not a stirng';
```

yes?

Reply:
Lack of standards across databases is so annoying. :)

Reply:
Yes.
It's annoying, right...

Reply:
Ok, removed highlighting of quoted identifers.